django/db/models/sql/query.py

```diff
@@ -1292,9 +1292,15 @@ def names_to_path(self, names, opts, allow_many=True, fail_on_missing=False):
             cur_names_with_path = (name, [])
             if name == 'pk':
                 name = opts.pk.name
+
+            field = None
             try:
                 field = opts.get_field(name)
+            except FieldDoesNotExist:
+                if name in self.annotation_select:
+                    field = self.annotation_select[name].field
```

Comment:
Please change to `field = self.annotation_select[name].output_field`. `output_field` is part of the expressions API. The field property is there for backwards compatibility with older code.